### PR TITLE
Add Freezing boundary conditions for Gh

### PR DESCRIPTION
**Pull request by prayush**
## Proposed changes

This adds freezing boundary conditions, where the characteristic projections of the RHS of evolution equations are set to 0 if the corresponding characteristic speed indicates an incoming mode.

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

**Comment by nilsvu** [comment]
@prayush how is this related to #3370? Would it be reasonable to add the feature as an input-file option to the existing class? Not making a suggestion yet, just asking to get an idea how things are related.

**Comment by prayush** [comment]
> @prayush how is this related to #3370? Would it be reasonable to add the feature as an input-file option to the existing class? Not making a suggestion yet, just asking to get an idea how things are related.

@nilsleiffischer in #3370 we are freezing the single mode `Vplus` while applying constraint preserving and physical corrections to the other 3 characteristic fields. This is the behavior of SpEC (see the link in the discussion under "proposed changes" in #3370 ) when one asks it to apply "Constraint preserving" boundary conditions. As Vplus should (almost) always be an outgoing mode, currently spectre simply leaves that mode unaltered. But to be completely consistent with SpEC, we need #3370 . The only time Vplus could possibly become incoming is either if the outer boundary is too close to the source or the domain itself is radially moving fast enough.

This PR, on the other hand, simply freezes all incoming modes (i.e. sets their time derivs to zero). This would be equivalent to the "Freezing" boundary condition in SpEC. It applies no correction terms whatsoever, and so we should preferably keep it separate from "ConstraintPreservingBjorhus".

**Comment by nilsvu** [comment]
@prayush thanks for explaining!

File: src/Evolution/Systems/GeneralizedHarmonic/BoundaryConditions/Freezing.hpp [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+      gsl::not_null<tnsr::aa<DataVector, Dim, Frame::Inertial>*>
+          dt_pi_correction,
+      gsl::not_null<tnsr::iaa<DataVector, Dim, Frame::Inertial>*>
+          dt_phi_correction,
+
+      const std::optional<tnsr::I<DataVector, Dim, Frame::Inertial>>&
+          face_mesh_velocity,
+      const tnsr::i<DataVector, Dim, Frame::Inertial>& normal_covector,
+      const tnsr::I<DataVector, Dim, Frame::Inertial>& /*normal_vector*/,
+      // c.f. dg_interior_evolved_variables_tags
```
Comment (geoffrey4444) [inline src/Evolution/Systems/GeneralizedHarmonic/BoundaryConditions/Freezing.hpp:99]: I don't understand these comments...please flesh them out

**Review by geoffrey4444 — CHANGES_REQUESTED** [review]
I took a look and didn't see anything significant to suggest in terms of code. 

However, before approving, I'd like to see a demonstration that some GH system can actually evolve stably with this BC...in my attempts, it doesn't work, but almost immediately leads to FPEs from constraint growth. Maybe I did something wrong in my tests, though?

My reasoning is that if the BC does not work, then it should not be merged. I could be talked out of that, though

**Comment by geoffrey4444** [comment]
Closed because we now have constraint preserving BCs for GH that work even with moving meshes.

**Pull request closed without merging**